numba/typing/npydecl.py

```diff
@@ -726,6 +726,28 @@ def generic(self, args, kws):
             return signature(restype, *args)
 
 
+@infer_global(numpy.linalg.inv)
+class LinalgInv(CallableTemplate):
+
+    def generic(self):
+        def typer(a):
+            if not isinstance(a, types.Array):
+                return
+            if not a.ndim == 2:
+                raise TypingError("np.linalg.inv() only supported on 2-D arrays")
+            if not a.layout in 'CF':
+                # Numpy seems to allow non-contiguous arguments, but we
+                # don't (we would need to copy before calling BLAS)
+                raise TypingError("np.linalg.inv() only supported on "
+                                  "contiguous arrays")
+            if not isinstance(a.dtype, (types.Float, types.Complex)):
+                raise TypingError("np.linalg.inv() only supported on "
+                                  "float and complex arrays")
+            return a.dtype
```

Comment:
This means the function returns a scalar the same type of the array, which doesn't look right. You probably want to return the array type itself (though perhaps always with the 'C' layout, which would be spelt `a.copy(layout='C')`).